src/circuit/gadget/ecc/chip/mul_fixed/base_field_elem.rs

```diff
@@ -0,0 +1,676 @@
+                    offset + idx,
+                    || z_next_val.ok_or(Error::SynthesisError),
+                )?;
+                CellValue::new(cell, z_next_val)
+            };
+
+            // Update `z`.
+            z = z_next;
+            running_sum.push(z);
+        }
+
+        Ok(EccBaseFieldElemFixed {
+            base_field_elem,
+            running_sum,
+        })
+    }
+}
+
+#[cfg(test)]
+pub mod tests {
+    use group::Curve;
+    use halo2::{
+        circuit::{Chip, Layouter},
+        plonk::Error,
+    };
+    use pasta_curves::{arithmetic::FieldExt, pallas};
+
+    use crate::circuit::gadget::{
+        ecc::{
+            chip::{EccChip, OrchardFixedBasesFull},
+            FixedPoint, Point,
+        },
+        utilities::{CellValue, UtilitiesInstructions},
+    };
+    use crate::constants;
+
+    pub fn test_mul_fixed_base_field(
+        chip: EccChip,
+        mut layouter: impl Layouter<pallas::Base>,
+    ) -> Result<(), Error> {
+        impl UtilitiesInstructions<pallas::Base> for EccChip {
+            type Var = CellValue<pallas::Base>;
+        }
+
+        // commit_ivk_r
+        let commit_ivk_r = OrchardFixedBasesFull::CommitIvkR;
+        test_single_base(
+            chip.clone(),
+            layouter.namespace(|| "commit_ivk_r"),
+            FixedPoint::from_inner(chip.clone(), commit_ivk_r),
+            commit_ivk_r.generator(),
+        )?;
+
+        // note_commit_r
+        let note_commit_r = OrchardFixedBasesFull::NoteCommitR;
+        test_single_base(
+            chip.clone(),
+            layouter.namespace(|| "note_commit_r"),
+            FixedPoint::from_inner(chip.clone(), note_commit_r),
+            note_commit_r.generator(),
+        )?;
+
+        // nullifier_k
+        let nullifier_k = OrchardFixedBasesFull::NullifierK;
+        test_single_base(
+            chip.clone(),
+            layouter.namespace(|| "nullifier_k"),
+            FixedPoint::from_inner(chip.clone(), nullifier_k),
+            nullifier_k.generator(),
+        )?;
+
+        // value_commit_r
+        let value_commit_r = OrchardFixedBasesFull::ValueCommitR;
+        test_single_base(
+            chip.clone(),
+            layouter.namespace(|| "value_commit_r"),
+            FixedPoint::from_inner(chip.clone(), value_commit_r),
+            value_commit_r.generator(),
+        )?;
+
+        // spend_auth_g
+        let spend_auth_g = OrchardFixedBasesFull::SpendAuthG;
+        test_single_base(
+            chip.clone(),
+            layouter.namespace(|| "spend_auth_g"),
+            FixedPoint::from_inner(chip, spend_auth_g),
+            spend_auth_g.generator(),
+        )?;
+
+        Ok(())
+    }
+
+    #[allow(clippy::op_ref)]
+    fn test_single_base(
+        chip: EccChip,
+        mut layouter: impl Layouter<pallas::Base>,
+        base: FixedPoint<pallas::Affine, EccChip>,
+        base_val: pallas::Affine,
+    ) -> Result<(), Error> {
```

Comment:
Why do these tests seem to be duplicated here and in `mul_fixed/full_width.rs`?

Reply:
They're duplicated because the `mul_fixed()` and `mul_fixed_base_field_elem()` are both usable with the same set of fixed points. Perhaps we could abstract these tests out?